ui/components/multichain/carousel/carousel.tsx

```diff
@@ -0,0 +1,231 @@
+      );
+    }
+
+    if (visibleSlides.length === 0) {
+      return null;
+    }
+
+    return (
+      <Box className="mm-carousel" ref={ref} {...(props as BoxProps<'div'>)}>
+        <ResponsiveCarousel
+          selectedItem={selectedIndex}
+          showArrows={false}
+          onChange={handleChange}
+          className="mm-carousel__carousel"
+          showStatus={false}
+          autoPlay={false}
+          swipeScrollTolerance={5}
+          centerSlidePercentage={getCenterSlidePercentage(visibleSlides.length)}
+          axis={'horizontal'}
+          preventMovementUntilSwipeScrollTolerance
+          emulateTouch
+          centerMode
+          swipeable
+        >
+          {visibleSlides.map((slide, index) => (
+            <BannerBase
+              data-testid={`slide-${slide.id}`}
+              onClick={() => {
+                if (index !== selectedIndex) {
+                  return;
+                }
+                if (slide.href) {
+                  global.platform.openTab({ url: slide.href });
+                }
+                if (slide.onClick) {
+                  slide.onClick();
+                }
+              }}
+              key={slide.id}
+              className="mm-carousel-slide"
+              startAccessory={
+                <img
+                  className="mm-carousel-slide__accessory"
+                  src={slide.image}
+                />
+              }
+              textAlign={TextAlign.Left}
+              alignItems={AlignItems.center}
+              title={t(slide.title)}
+              description={t(slide.description)}
+              titleProps={{
+                variant: TextVariant.bodySmMedium,
+                fontWeight: FontWeight.Medium,
+                marginLeft: 2,
+              }}
+              backgroundColor={BackgroundColor.backgroundAlternative}
+              borderColor={BorderColor.infoMuted}
+              borderWidth={1}
+              descriptionProps={{
+                variant: TextVariant.bodyXs,
+                fontWeight: FontWeight.Normal,
+                marginLeft: 2,
+              }}
+              onClose={
+                Boolean(handleClose) && !slide.undismissable
+                  ? (e: React.MouseEvent<HTMLElement>) =>
+                      handleClose(e, slide.id)
+                  : undefined
+              }
+              closeButtonProps={{
+                className: 'mm-carousel-slide__close-button',
+              }}
+              style={{
+                height: BANNER_STYLES.HEIGHT,
+                margin: getSlideMargin(index, visibleSlides.length),
+                width: getSlideWidth(visibleSlides.length),
+                position: 'relative',
+              }}
+              padding={0}
+              paddingLeft={3}
+              paddingRight={3}
+            />
+          ))}
+        </ResponsiveCarousel>
+      </Box>
+    );
+  },
+);
+
+function getSlideMargin(index: number, totalSlides: number) {
```

Comment:
Can we move it to a selector file?

Reply:
What do you mean by selector file? These margins are specific to the carousel component

Reply:
I mean moving this function to a different file and keeping the carousel a dumb component to avoid any function logic here. We can do it in a follow up PR